src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -686,6 +686,90 @@ namespace OpenLoco::Vehicles
         return AirportObjectFlags::acceptsHeavyPlanes;
     }
 
+    // 0x004AFFF3
+    VehicleBogie* VehicleBogie::flipCar()
+    {
+        Vehicle train(head);
+        VehicleBase* precedingCarBody = train.veh2;
+        while (precedingCarBody->nextVehicleComponent()->asBase<VehicleBogie>() != this)
+        {
+            precedingCarBody = precedingCarBody->nextVehicleComponent();
+        }
+
+        VehicleBase* ptr = this; // lvalue required
```

Comment:
what i normally do is:
```cpp
for(auto& car : train.cars)
{
    if (car.front == this)
    {
       for(auto& carComponent : car){
          // do car component stuff
       }
    }
}
```
As you have worked out I hadn't designed the car components for construction outside of the train so its not very neat.